### ISSUE TITLE
Construct the significant terms aggregation builder's parser statically

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#25519]
Right now we build a new ObjectParser every time we want get a significant terms aggregation. We do this because we had to pass in `ParseFieldRegistry<SignificanceHeuristicParser>`. I think we can avoid this by moving `ParseFieldRegistry<SignificanceHeuristicParser>` to `NamedXContentRegistry` style parsing.

**Comment by nik9000** [comment]
Marked a `low hanging fruit` because it should be *fairly* simple to do but big enough that it certainly deserves its own PR.

**Comment by cbuescher** [comment]
Same is true for SignificantTextAggregationBuilder which follows the same pattern. I took a quick first look at it, changing the way the heuristics are registered is only part of this. I quickly got stuck when trying to convert the part around https://github.com/elastic/elasticsearch/blob/f576c987ce2615f77a8de75741b0f5448229805f/core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java#L98 because currently each potential heuristic is declared on the ObjectParser using its name. I think we might need something like what I started in https://github.com/elastic/elasticsearch/pull/24020 to be able to do this with ObjectParser, most other places we parse named things like this we still do it in the "classic" manual parsing style. Just adding this note here since this issue might be a little more involved than estimated.
